ol/tower/src/backlog.rs

```diff
@@ -1,26 +1,43 @@
 //! Miner resubmit backlog transactions module
 #![forbid(unsafe_code)]
 
-use cli::{diem_client::DiemClient};
-use ol_types::block::VDFProof;
-use txs::submit_tx::{TxParams, eval_tx_status};
-use std::{fs::File, path::PathBuf, thread, time};
-use ol_types::config::AppCfg;
 use crate::commit_proof::commit_proof_tx;
-use std::io::BufReader;
 use crate::proof::{parse_block_height, FILENAME};
-use anyhow::{bail, Result, Error};
+use anyhow::{bail, Error, Result};
+use cli::diem_client::DiemClient;
+use diem_client::views::TowerStateResourceView;
 use diem_logger::prelude::*;
+use ol_types::block::VDFProof;
+use ol_types::config::AppCfg;
+use std::io::BufReader;
+use std::{cmp::max, fs::File, path::PathBuf};
+use txs::epoch::get_epoch;
+use txs::submit_tx::{eval_tx_status, TxParams};
 
-/// Submit a backlog of blocks that may have been mined while network is offline. 
-/// Likely not more than 1. 
+// only 72 proofs are allowed per epoch
+const MAX_PROOFS_PER_EPOCH: i64 = 72;
```

Comment:
Can this be fetched from the move globals so that we don't have duplication?

Reply:
My preference is to catch this from the error code which Move throws when the upper bound is reached. `130108`
https://github.com/OLSF/libra/blob/main/language/diem-framework/modules/0L/TowerState.move#L278-L281

More importantly we're missing some error maps of 0L specific errors which the client should identify. See this draft issue I opened. https://github.com/OLSF/libra/issues/933

Reply:
I will fetch it from global store

Reply:
actually after some thinking. I don't think this approach works because with my change tower backlog submission already aborts on any error from submission. So it doesn't matter error code we still should abort submission. This is to solve problem #2 I found with the tower.
And your suggestion to watch for specific error doesn't solve problem #1 in tower - we should not call backlog submission at all if we reached max number of proofs in epoch. Otherwise each tower will keep sending at least 1 proof every 30 mins causing move abort.

Reply:
The thing to keep in mind is that MAX_PROOFS_PER_EPOCH will not always be 72.  And the users need to have the tower continue to make proofs if there's trouble connecting to network or submitting proofs (tower app should not exit).

Reply:
how do I read number 72 from move config? Is there a code snipped to do that? When backup returns (success or failure) , tower proceeds to mine the next proof which  will attempt to be submitted in the next invocation of backlog and so on

Reply:
We have a function in Move which fetches this hardcoded value, but we don't keep it stored in state. So actually we may not be able to get it dynamically through usual APIs

Reply:
I think it's fine to merge this PR, even if we then will have the allowed proofs per epoch configured at two places. Given the high complexity to make it perfect....